qiita_db/base.py

```diff
@@ -128,28 +130,34 @@ def _check_id(self, id_):
         the other classes. However, still defining here as there is only one
         subclass that doesn't follow this convention and it can override this.
         """
-        self._check_subclass()
-
-        conn_handler = SQLConnectionHandler()
-
-        return conn_handler.execute_fetchone(
-            "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
-            "{0}_id=%s)".format(self._table), (id_, ))[0]
+        sql = "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE {0}_id=%s)".format(
+            self._table)
+        trans.add(sql, [id_])
+        return trans.execute()[-1][0][0]
 
-    def __init__(self, id_):
+    def __init__(self, id_, trans=None):
         r"""Initializes the object
 
         Parameters
         ----------
-        id_: the object identifier
+        id_: object
+            The object identifier
+        trans: Transaction
+            Transaction in which this method should be executed
 
         Raises
         ------
         QiitaDBUnknownIDError
             If `id_` does not correspond to any object
         """
-        if not self._check_id(id_):
-            raise QiitaDBUnknownIDError(id_, self._table)
+        self._check_subclass()
+
+        trans = trans if trans is not None else Transaction(
```

Comment:
this should be a method. what about the following signature:

```python
trans = Transaction.create_if_none(trans, self, id_)
```

The code could look something like:

```python
@classmethod
def create_if_none(cls, trans, obj, id_):
    if trans is None:
        return trans
    else:
        return cls("init_%s_%s" % (obj.__class__.__name__, id_))
```